### PR TITLE
Add freeimage/3.18.0

### LINKED ISSUE #3027: [request] freeimage/3.18.0
### Package Details
  * Package Name/Version: freeimage/3.18.0
  * Website: https://freeimage.sourceforge.io/index.html
  * Source code: http://downloads.sourceforge.net/freeimage/FreeImage3180.zip

### Description Of The Library / Tool
Image file reading and writing

### PR DESCRIPTION
**Pull request by pbreaux**
Specify library name and version:  **freeimage/3.18.0**

This is a new addition to the ConanCenter.

closes #3027

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/freeimage/all/test_package/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake", "cmake_find_package"
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self.settings):
```
Comment (intelligide) [inline recipes/freeimage/all/test_package/conanfile.py:15]:
```suggestion
        if not tools.cross_building(self):
```

File: recipes/freeimage/all/test_package/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
```
Comment (intelligide) [inline recipes/freeimage/all/test_package/conanfile.py:6]:
```suggestion
     settings = "os", "arch", "compiler", "build_type"
```

File: recipes/freeimage/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+import shutil
+
+
+class FreeImageConan(ConanFile):
+    name = "freeimage"
+    description = "Open Source library project for developers who would like to support popular graphics image formats"\
+                  "like PNG, BMP, JPEG, TIFF and others as needed by today's multimedia applications."
+    homepage = "https://freeimage.sourceforge.io"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "FreeImage", "GPL-3.0-or-later", "GPL-2.0-or-later", "FreeImage"
```
Comment (intelligide) [inline recipes/freeimage/all/conanfile.py:12]:
```suggestion
    license = "FreeImage", "GPL-3.0-or-later", "GPL-2.0-or-later"
```

File: recipes/freeimage/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        tools.check_min_cppstd(self, "11")
+        if self.options.shared:
+            del self.options.fPIC
+        self.output.warn("G3 plugin and JPEGTransform are disabled.")
+        if bool(self.options.with_jpeg):
```
Comment (intelligide) [inline recipes/freeimage/all/conanfile.py:63]:
```suggestion
        if self.options.with_jpeg is not None:
```

File: recipes/freeimage/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+            self.requires("libraw/0.20.2")
+        if self.options.with_jxr:
+            self.requires("jxrlib/cci.20170615")
+        if self.options.with_tiff:
+            self.requires("libtiff/4.3.0")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "FreeImage"
+        os.rename(extracted_dir, self._source_subfolder)
```
Comment (intelligide) [inline recipes/freeimage/all/conanfile.py:91]:
```suggestion
        tools.get(**self.conan_data["sources"][self.version],
                  destination=self._source_subfolder, strip_root=True)
```

File: recipes/freeimage/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,160 @@
+from conans import ConanFile, CMake, tools
+import os
+import shutil
+
+
```
Comment (intelligide) [inline recipes/freeimage/all/conanfile.py:5]:
```suggestion

required_conan_version = ">=1.43.0"


```

File: recipes/freeimage/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+class FreeImageConan(ConanFile):
+    name = "freeimage"
+    description = "Open Source library project for developers who would like to support popular graphics image formats"\
+                  "like PNG, BMP, JPEG, TIFF and others as needed by today's multimedia applications."
+    homepage = "https://freeimage.sourceforge.io"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "FreeImage", "GPL-3.0-or-later", "GPL-2.0-or-later", "FreeImage"
+    topics = ("conan", "freeimage", "image", "decoding", "graphics")
```
Comment (intelligide) [inline recipes/freeimage/all/conanfile.py:13]:
```suggestion
    topics = ("freeimage", "image", "decoding", "graphics")
```

File: recipes/freeimage/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+    exports_sources = ["CMakeLists.txt", "patches/*"]
```
Comment (intelligide) [inline recipes/freeimage/all/conanfile.py:14]:
```suggestion
```

File: recipes/freeimage/all/conanfile.py [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
```
Comment (intelligide) [inline recipes/freeimage/all/conanfile.py:53]:
```suggestion

    def export_sources(self):
        self.copy("CMakeLists.txt")
        for patch in self.conan_data.get("patches", {}).get(self.version, []):
            self.copy(patch["patch_file"])

```

**Review by intelligide — CHANGES_REQUESTED** [review]

**Review by uilianries — APPROVED** [review]
- FreeImage does not have an official cmake file, but it refers the project as FreeImage, so I'm fine accepting it as cmake target
- The upstream offers a Makefile, which didn't work to me at first try with GCC11. So I'm fine accepting this cmake wrapper
- I see a big number of patches, and they are related to external dependencies only, so I'm fine with it.

LGTM

**Review by pbreaux — COMMENTED** [review]
Changes applied.

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

File: recipes/freeimage/all/conanfile.py [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+                components.append("openexr::openexr")
+            if self.options.with_raw:
+                components.append("libraw::libraw")
+            if self.options.with_jxr:
+                components.append("jxrlib::jxrlib")
+            if self.options.with_tiff:
+                components.append("libtiff::libtiff")
+            return components
+
+        self.cpp_info.names["pkg_config"] = "freeimage"
```
Comment (prince-chrismc) [inline recipes/freeimage/all/conanfile.py:156]: Missing new generators 🤷

**Review by prince-chrismc — APPROVED** [review]
Great work !!

**Review by intelligide — APPROVED** [review]

**Pull request merged**